mantidproject/mantid
# DirectILL input numors in the SampleLogs

This PR partially addresses the issue of retrieving the ids (numors) of files that were used to create final output data in the Direct ILL reduction workflow. It adds a new sample log: 'run_list' to two NeXus loaders handling ILL direct geometry instruments: `LoadILLTOF2` and `LoadILLDiffraction` algorithms to ease the extraction of numors used to load the data. The information containing numors of correction workspaces will be added in a separate reduction algorithm, currently in development.



**To test:**

1. Load one or more data files using either directly `LoadAndMerge` or indirectly through `DirectILLCollectData`, for example using PANTHER's unit test data under: Data/UnitTest/ILL/PANTHER
2. Observe if the SampleLog 'run_list' is created and the value contains all the file names in your input.

Partially fixes #31883 . 


Release note added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
